core/src/main/java/org/axonframework/config/DefaultConfigurer.java

```diff
@@ -216,6 +218,22 @@ protected QueryBus defaultQueryBus(Configuration config) {
                                   config.getComponent(QueryInvocationErrorHandler.class));
     }
 
+    /**
+     * Provides the default QueryUpdateEmitter implementation. Subclasses may override this method to provide their own
+     * default.
+     *
+     * @param config The configuration based on which the component is initialized
+     * @return The default QueryUpdateEmitter to use
+     */
+    protected QueryUpdateEmitter defaultQueryUpdateEmitter(Configuration config) {
+        QueryBus queryBus = config.getComponent(QueryBus.class);
```

Comment:
You are right! We should totally check for `QueryUpdateEmitter` in configuration components. Nice catch.

Reply:
You are right! We should definitely check for `QueryUpdateEmitter` in the configuration. Nice catch.

Reply:
I gave it a second thought (read this: tried to implement it and saw why it cannot be done). This is the default `queryUpdateEmitter` which should be the same as query bus. If someone wants to override it, it can be achieved by `configurer.registerComponent`. 

Regarding the message: the exception is thrown if someone configured the query bus which does not support emitting. So, the message says "register the query bus which is capable to emit updates", or "register separate query update emitter". IMO, nothing should be changed.

On the other hand, @abuijze and I had a discussion whether `QueryBus` should extend `QueryUpdateEmitter`, and the argument against it was that this way the readability is better - components which are providing the updates (probably) will not need the query bus, hence only `QueryUpdateEmitter` is sufficient.

Reply:
Aah of course, as this is the default version which will be both an emitter and bus, this makes more sense.

In the last part of your comment you elaborate on a discussion you had with @abuijze about having the `QueryBus extends QuerryUpdateEmitter`: I had that same hunch initially, asking my self why you didn't do that. I kinda follow the stance you've taken, but your point of components-that-wont-need-the-bus-just-the-emitter could still hold if you'd have the `QueryBus` interface extend the `QueryUpdateEmitter`, right?

And another question: will any future `QueryBus` implementation we'd add also implement both interfaces (for example the `DistributedQueryBus`)? I'd assume so, but still would like to hear a potential thought process behind that.

Reply:
I guess that all query buses will be emitters also, but didn't want to make a prediction there, that's why I left it separated. If you think that it's going to be always the case, we can easily make the change :).